test/c_src/finest.cpp

```diff
@@ -74,6 +76,80 @@ struct ExError {
   static constexpr auto is_exception = true;
 };
 
+namespace __private__ {
+class : public std::pmr::memory_resource {
+private:
```

Comment:
Honestly, if we only pass the STL `Alloc` parameter through, I don't think we need extra tests with the custom allocator implementation here.

Reply:
I think the same point applies to `struct Allocator`?

Reply:
I was thinking that's fine, but I guess the extra test doesn't hurt either and the allocator is pretty straightforward, so let's keep that, yeah :)